packages/timeline-state-resolver/src/service/DeviceInstance.ts

```diff
@@ -242,6 +242,13 @@ export class DeviceInstanceWrapper extends EventEmitter<DeviceInstanceEvents> {
 			},
 
 			resetState: async () => {
+				await this._stateHandler.setCurrentState(undefined)
+				await this._stateHandler.clearFutureStates()
+				this.emit('resetResolver')
+			},
+
+			resetToState: async (state: any) => {
+				await this._stateHandler.setCurrentState(state)
 				await this._stateHandler.clearFutureStates()
 				this.emit('resetResolver')
```

Comment:
This isn't something to fix now, but doesn't calling `this.emit('resetResolver')` result in clearing the state of every device and rebuilding it all from the timeline again?
That feels like a rather heavy operation to do whenever a device manages to connect, especially as the states returned by the resolver will not be different to before.
Perhaps something in each service could cache the states as given by the resolver, so that `resetState` and `resetToState` can reuse those instead?